### DISCUSSION TITLE
Scripting: Can the timer sections execute calculations/functions?

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#21888] [Q&A]
I am using two timers as follows to display alternating info on a TFT display

```
>ti1
print t1:
ts2(10000)
dt ...

>ti2
print t2:
ts1(10000)
dt ...

```
Above works fine.

However when I tried to introduce some string manipulation inside it does not seem to work

```
>ti1
print t1:
ts2(10000)
dt ...

>ti2
print t2:
ts1(10000)
mstr=sb(mstr 1 5)
print t2: %mstr%
dt ...

```

Am I trying to do something which is not supported inside the` >ti2` section?

PS:
FYI, similar commands in `>E` section works fine
`mstr=sb(mstr 1 sl(mstr)-2)`

**Comment by gemu2015** [comment]
no restriction, just tested and works.
however copying the string to itself like in your example will only work once

**Reply by jkwim** [reply]
I tried this but no success:
Tried to extract a substring in to `tstr`.

```
 if (sl(mstr)>13) {
  print t2: length>13 %mstr%
  tstr=sb(tstr 1 13)
  print t2: %tstr%
  dt [p13]%tstr%
  tstr=sb(mstr 14 sl(mstr))
  print t2: %tstr%
  dt [y210p13]%tstr%
  }
```

**Reply by gemu2015** [reply]
tstr=sb(tstr 1 13) should be tstr=sb(mstr 1 13)

**Reply by jkwim** [reply]
Thanks. That solved the problem.

BTW can the 'dt' command, or for that matter any command in a script, accept an expression which will be evaluated as a parameter value.

For example :

`dt [y210p13]sb(tstr 1 13)`

**Reply by gemu2015** [reply]
you can insert any variable (numeric or string) with the %var%  (var = var name)
you can also do math with numeric variables by putting the expression into brackets %(var1*var2)%
you may not execute script cmds like sb()

**Reply by jkwim** [reply]
Got it!

Thank!
